packages/superset-ui-number-format/src/factories/createD3NumberFormatter.ts

```diff
@@ -7,14 +7,18 @@ export default function createD3NumberFormatter(config: {
   description?: string;
   formatString: string;
   label?: string;
+  locale?: FormatLocaleDefinition;
 }) {
-  const { description, formatString = isRequired('config.formatString'), label } = config;
+  const { description, formatString = isRequired('config.formatString'), label, locale } = config;
 
   let formatFunc: NumberFormatFunction;
   let isInvalid = false;
 
   try {
-    formatFunc = d3Format(formatString);
+    formatFunc =
+      typeof locale === 'undefined'
```

Comment:
why not just `locale === undefined`? or even just `locale ? formatLocale(locale).format(formatString) : d3Format(formatString)`

Reply:
I have been adopting this pattern from @williaster. Kinda forgot the original rationale.

Reply:
> because `undefined` is not a keyword, and can be redefined (in any scope but global in ES5+)